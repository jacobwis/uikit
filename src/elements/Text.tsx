import styled from 'styled-components';
import { FontSize, FontWeight, Leading } from '../theme/typography';

export interface TextProps {
  size?: FontSize;
  leading?: Leading;
  weight?: FontWeight;
}

const Text = styled.span<TextProps>`
  color: ${({ theme }) => theme.colors.gray[900]};
  font-size: ${({ theme, size = 'base' }) => theme.typography.size[size]};
  font-weight: ${({ theme, weight = 'normal' }) =>
    theme.typography.weight[weight]};
  line-height: ${({ theme, size = 'base', leading }) => {
    if (leading) {
      return theme.typography.leading[leading];
    }

    switch (size) {
      case 'xs':
      case '2xl':
      case '3xl':
      case '4xl':
        return theme.typography.leading.tight;
      case '5xl':
      case '6xl':
        return theme.typography.leading.none;
      default:
        return theme.typography.leading.normal;
    }
  }};
`;

export default Text;
