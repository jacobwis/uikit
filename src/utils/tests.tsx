import * as React from 'react';
import { render, RenderOptions, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../theme';

const AllProviders: React.FC = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
): RenderResult => render(ui, { wrapper: AllProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
