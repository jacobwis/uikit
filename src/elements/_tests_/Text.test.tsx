import * as React from 'react';
import { screen, render } from '../../utils/tests';
import Text from '../Text';

const sizes = [
  'xs',
  'sm',
  'base',
  'lg',
  'xl',
  '2xl',
  '3xl',
  '4xl',
  '5xl',
  '6xl',
];

describe('<Text />', () => {
  it('should render a "span" element by default', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text>{testText}</Text>);
    expect(screen.getByText(testText).tagName).toEqual('SPAN');
  });

  it('should match the snapshot when "props.size" equals "xs"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="xs">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "sm"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="sm">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "base"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="base">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "lg"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="lg">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "2xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="2xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "3xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="3xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "4xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="4xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "5xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="5xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.size" equals "6xl"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text size="6xl">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.leading" equals "none"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text leading="none">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.leading" equals "tight"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text leading="tight">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.leading" equals "normal"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text leading="normal">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.leading" equals "loose"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text leading="loose">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.weight" equals "normal"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text weight="normal">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.weight" equals "bold"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text weight="bold">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });

  it('should match the snapshot when "props.weight" equals "extrabold"', () => {
    const testText = 'This is the text for the Text component.';
    render(<Text weight="extrabold">{testText}</Text>);
    expect(screen.getByText(testText)).toMatchSnapshot();
  });
});
