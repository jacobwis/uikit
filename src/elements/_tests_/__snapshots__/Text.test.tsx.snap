// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match the snapshot when "props.leading" equals "loose" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 2;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.leading" equals "none" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.leading" equals "normal" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.leading" equals "tight" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.25;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "2xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 1.5rem;
  font-weight: 400;
  line-height: 1.25;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "3xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 1.875rem;
  font-weight: 400;
  line-height: 1.25;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "4xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 2.25rem;
  font-weight: 400;
  line-height: 1.25;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "5xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 3rem;
  font-weight: 400;
  line-height: 1;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "6xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 3.75rem;
  font-weight: 400;
  line-height: 1;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "base" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "lg" 1`] = `
.c0 {
  color: #111827;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "sm" 1`] = `
.c0 {
  color: #111827;
  font-size: 0.875rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "xl" 1`] = `
.c0 {
  color: #111827;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.size" equals "xs" 1`] = `
.c0 {
  color: #111827;
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1.25;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.weight" equals "bold" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 600;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.weight" equals "extrabold" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 800;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;

exports[`<Text /> should match the snapshot when "props.weight" equals "normal" 1`] = `
.c0 {
  color: #111827;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1.5;
}

<span
  class="c0"
>
  This is the text for the Text component.
</span>
`;
